src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java

```diff
@@ -118,5 +121,17 @@ public CredentialSupport getCredentialSupport(Class<?> credentialType) {
             final Password password = map.get(principal);
             return credentialType.isInstance(password) ? credentialType.cast(password) : null;
         }
+
+        public VerificationResult verifyCredential(final Object credential) {
```

Comment:
It is the possibly that complicates it the most, something could be definitely verifiable but only possibly obtainable.